src/server/mod.rs

```diff
@@ -204,11 +204,27 @@ where S: NetworkStream + Clone, H: Handler {
         }
 
         keep_alive = http::should_keep_alive(req.version, &req.headers);
-        let mut res = Response::new(&mut wrt);
-        res.version = req.version;
-        handler.handle(req, res);
-        debug!("keep_alive = {:?}", keep_alive);
+        let version = req.version;
```

Comment:
This function is starting to get *really* long, can you split this section into another fn?